modules/data.land/R/extract_soil_nc.R

```diff
@@ -15,35 +15,38 @@
 #'    PEcAn.data.land::extract_soil_gssurgo(outdir,lat,lon)
 #' }
 #' @author Hamze Dokoohaki
-
-extract_soil_gssurgo<-function(outdir,lat,lon){
+#' 
+extract_soil_gssurgo<-function(outdir, lat, lon){
   #reading the mapunit based on latitude and longitude of the site
   #the output is a gml file which need to be downloaded and read as a spatial file but I don't do that.
   #I just read the file as a text and parse it out and try to find the mukey==mapunitkey
-  RCurl::getURL(paste0("https://sdmdataaccess.nrcs.usda.gov/Spatial/SDMWGS84Geographic.wfs?SERVICE=WFS&VERSION=1.1.0&REQUEST=GetFeature&TYPENAME=MapunitPoly&FILTER=<Filter><DWithin><PropertyName>Geometry</PropertyName><gml:Point>%20<gml:coordinates>",
-                lon ,",", lat,"</gml:coordinates></gml:Point><Distance%20units=%27m%27>0</Distance>%20</DWithin></Filter>"), ssl.verifyhost=FALSE, ssl.verifypeer=FALSE)->xmll
-  regexpr('<ms:mukey>', xmll)->startp
-  regexpr('</ms:mukey>', xmll)->stopp
+  xmll <- RCurl::getURL(paste0("https://sdmdataaccess.nrcs.usda.gov/Spatial/SDMWGS84Geographic.wfs?SERVICE=WFS&VERSION=1.1.0&REQUEST=GetFeature&TYPENAME=MapunitPoly&FILTER=<Filter><DWithin><PropertyName>Geometry</PropertyName><gml:Point>%20<gml:coordinates>",
+                lon ,",", lat,"</gml:coordinates></gml:Point><Distance%20units=%27m%27>0</Distance>%20</DWithin></Filter>"), ssl.verifyhost=FALSE, ssl.verifypeer=FALSE)
+  
+  startp <- regexpr('<ms:mukey>', xmll)
+  stopp <- regexpr('</ms:mukey>', xmll)
+  
   #if you found the mapunit key
   if (startp==-1 | stopp==-1) PEcAn.logger::logger.error("There was no mapunit keys found for this site.")
     # caaling the query function sending the mapunit key
-    soilprop<-gSSURGO.Query(substr(xmll, startp%>%as.numeric+10, stopp%>%as.numeric-1))
+    soilprop <- gSSURGO.Query(substr(xmll, startp%>%as.numeric+10, stopp%>%as.numeric-1))
```

Comment:
Are pipes here _really_ necessary? I'm generally all for pipes, but I find `startp %>% as.numeric + 10` to be more confusing than just `as.numeric(startp) + 10`.